### PR TITLE
Ws 016 add whale frontend

### PR DESCRIPTION
**Pull request by charlottemothersole**

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.jar
Comment (TimLeach635) [inline gradle/wrapper/gradle-wrapper.jar:1]: These gradle files shouldn't be included in the PR! Please remove them (let me know if you can't remember how!)

File: gradle/wrapper/gradle-wrapper.properties
Comment (TimLeach635) [inline gradle/wrapper/gradle-wrapper.properties:1]: Shouldn't be in this PR

File: gradlew
Comment (TimLeach635) [inline gradlew:1]: Shouldn't be in this PR

File: gradlew.bat
Comment (TimLeach635) [inline gradlew.bat:1]: Shouldn't be in this PR

File: src/main/java/com/whalespottingjava/controllers/SightingController.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        return "sighting_test";
+    }
+
+    //for admin use
+    @GetMapping("/PendingSightings")
+    public String getPendingSightings() {
+        return "pending_sightings";
+    }
+
+    //For user use
```
Comment (TimLeach635) [inline src/main/java/com/whalespottingjava/controllers/SightingController.java]: I'm not sure these comments are necessary, but in any case, prefer to have the same casing between them, and have a space after the slashes, like so:
```suggestion
    // For admin use
    @GetMapping("/PendingSightings")
    public String getPendingSightings() {
        return "pending_sightings";
    }

    // For user use
```

File: src/main/java/com/whalespottingjava/controllers/SightingController.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    public SightingController(SightingService sightingService) {
+        this.sightingService = sightingService;
+    }
+
+    @GetMapping("/add-whale-sighting")
+    public String getAddSightingPage() {
+        return "add_sighting";
+    }
+
+    @GetMapping("/Sightings")
```
Comment (TimLeach635) [inline src/main/java/com/whalespottingjava/controllers/SightingController.java]: Prefer lowercase web routes
```suggestion
    @GetMapping("/sightings")
```

File: src/main/java/com/whalespottingjava/controllers/SightingController.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    public String getAllSighting(Model model) {
```
Comment (TimLeach635) [inline src/main/java/com/whalespottingjava/controllers/SightingController.java]: Correct plural
```suggestion
    public String getAllSightings(Model model) {
```

File: src/main/java/com/whalespottingjava/controllers/SightingController.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        model.addAttribute("sightings", sightingService.getAllSightings());
+        return "sighting_test";
+    }
+
+    //for admin use
+    @GetMapping("/PendingSightings")
```
Comment (TimLeach635) [inline src/main/java/com/whalespottingjava/controllers/SightingController.java]: Prefer `kebab-case` (it's called that because it looks like the letters are being skewered on a shish kebab) for web routes
```suggestion
    @GetMapping("/pending-sightings")
```

File: src/main/java/com/whalespottingjava/controllers/SightingController.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    public String getPendingSightings() {
+        return "pending_sightings";
+    }
+
+    //For user use
+    @GetMapping("/ApprovedSightings")
```
Comment (TimLeach635) [inline src/main/java/com/whalespottingjava/controllers/SightingController.java]: As above, prefer kebab case
```suggestion
    @GetMapping("/approved-sightings")
```

File: src/main/java/com/whalespottingjava/models/database/Sighting.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+    protected Sighting() {}
+    
+    public Sighting(Date date, double latitude, double longitude) {
+        this.date = date;
+        this.latitude = latitude;
+        this.longitude = longitude;
+        this.approved = false;
+    }
+
+
```
Comment (TimLeach635) [inline src/main/java/com/whalespottingjava/models/database/Sighting.java:41]: Unnecessary blank line
```suggestion
```

File: src/main/java/com/whalespottingjava/models/database/Sighting.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = ID_SEQUENCE)
+    private Long id;
+
+    @Column(nullable = false)
+    private int member_id;
+
+    private Date date;
+    private double latitude;
+    private double longitude;
+    private Boolean approved;
```
Comment (TimLeach635) [inline src/main/java/com/whalespottingjava/models/database/Sighting.java:23]: Prefer booleans to be named with "is" or "has" or some other thing that makes it seem like a statement
```suggestion
    private Boolean isApproved;
```
(You'll need to change this throughout the file)

File: src/main/java/com/whalespottingjava/services/SightingService.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+import java.util.List;
+
+@Service
+public class SightingService {
+
+    private final SightingRepository sightingRepository;
+
+    @Autowired
+    public SightingService(SightingRepository sightingRepository) {this.sightingRepository = sightingRepository;}
```
Comment (TimLeach635) [inline src/main/java/com/whalespottingjava/services/SightingService.java]: Prefer this to be indented
```suggestion
    public SightingService(SightingRepository sightingRepository) {
        this.sightingRepository = sightingRepository;
    }
```

File: src/main/resources/static/css/addSighting.css [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    margin: 0 auto;
+}
+
+form > input {
+    margin: 10px 0;
+    padding: 7px 5px;
+    border: none;
+    border-radius: 5px;
+    max-width: 175px;
+
```
Comment (TimLeach635) [inline src/main/resources/static/css/addSighting.css]: Unnecessary blank line
```suggestion
```

File: src/main/resources/static/css/addSighting.css [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+        margin: 0 0 20px 0;
+    }
+
+    #add-sighting {
+        width: unset;
+        margin: 0;
+    }
+}
+
+
```
Comment (TimLeach635) [inline src/main/resources/static/css/addSighting.css]: Exactly one blank line at end of file
```suggestion
```

File: src/main/resources/static/css/application.css [hunk 11]
```diff
@@ -33,11 +40,9 @@
     background-color: rgba(27, 61, 175, 0.849);
     bottom: 0;
     display: flex;
-
     flex-direction: row;
     list-style: none;
     padding-left: 1rem;
-
```
Comment (TimLeach635) [inline src/main/resources/static/css/application.css]: Avoid making changes, especially stylistic ones, in areas outside of where you're working! Please undo these line deletions

File: src/main/resources/templates/add_sighting.html [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+                <input type="number" required placeholder="e.g. 38.8951"/>
+                <label for="longitude">
+                    Longitude
+                </label>
+                <input type="number" required placeholder="e.g. -77.0364"/>
+                <input class="button" type="submit" value="Add Sighting"/>
+            </form>
+        </div>
+    </body>
+</html>
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: Exactly one blank line at end of file
```suggestion
</html>
```

File: src/main/resources/templates/add_sighting.html [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+                <input type="date" required/>
+                <label for="latitude">
+                    Latitude
+                </label>
+                <input type="number" required placeholder="e.g. 38.8951"/>
+                <label for="longitude">
+                    Longitude
+                </label>
+                <input type="number" required placeholder="e.g. -77.0364"/>
+                <input class="button" type="submit" value="Add Sighting"/>
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: Prefer space before `/>` in self-closing HTML tags
```suggestion
                <input type="number" required placeholder="e.g. 38.8951" />
                <label for="longitude">
                    Longitude
                </label>
                <input type="number" required placeholder="e.g. -77.0364" />
                <input class="button" type="submit" value="Add Sighting" />
```

File: src/main/resources/templates/add_sighting.html [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+    </head>
+    <body>
+        <div layout:fragment="content">
+            <h1 id="add-sighting-header">Add a whale sighting</h1>
+            <p id="add-sighting-details">Please add the details of your sighting below.</p>
+            <form id="add-sighting" action="" method="POST">
+                <label for="date">
+                    Date of Sighting
+                </label>
+                <input type="date" required/>
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: As below
```suggestion
                <input type="date" required />
```

File: src/main/resources/templates/add_sighting.html [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+                <label for="latitude">
+                    Latitude
+                </label>
+                <input type="number" required placeholder="e.g. 38.8951"/>
```
Comment (TimLeach635) [inline src/main/resources/templates/add_sighting.html]: You've used the `for` field, but for this to work, you need to also set the ID of the input field to the same thing
```suggestion
                <label for="latitude">
                    Latitude
                </label>
                <input id="latitude" type="number" required placeholder="e.g. 38.8951"/>
```

Depending on how this wires up elsewhere, you may also need to add the `name` attribute to these input elements

Comment (TimLeach635) [reply src/main/resources/templates/add_sighting.html]: Please make this change on the other inputs too!

**Review by TimLeach635 — DISMISSED** [review]
Nice! Just some changes to make

**Comment by charlottemothersole** [comment]
All changes now made and ready for review

**Comment by AndrewYHuang** [comment]
> All changes now made and ready for review

I think there might be a few more changes that you might need to make or push - there's a few hidden conversations that haven't been resolved yet, and I think the gradle files have been un-reverted again!

File: src/main/java/com/whalespottingjava/models/database/Sighting.java (hunk 7 above)
Comment (AndrewYHuang) [reply src/main/java/com/whalespottingjava/models/database/Sighting.java:23]: Plan to fix in a separate PR

**Review by AndrewYHuang — APPROVED** [review]
